pageserver/src/storage_sync/index.rs

```diff
@@ -288,8 +368,11 @@ impl IndexPart {
         metadata_bytes: Vec<u8>,
     ) -> Self {
         Self {
+            version: Self::LATEST_VERSION,
             timeline_layers,
             missing_layers,
+            // TODO: for tests this could be easily set
+            layer_metadata: Default::default(),
```

Comment:
I've lamented about this from type inference standpoint, but even from PR readability one, I cannot understand what's written there, compared to `T::default()`.

Reply:
Should be already fixed.